pkg/events/registry.go

```diff
@@ -0,0 +1,37 @@
+package events
+
+import (
+	"k8s.io/klog"
+)
+
+type EventRegistry struct {
+	eventHandlers map[string]EventHandler
+	name          string
+}
+
+func NewRegistry(name string) EventRegistry {
+	return EventRegistry{
+		name: name,
+	}
+}
+
+// Adds a supported driver, prints a fatal error in the case of double registration
+func (er *EventRegistry) AddDriver(name string, eventHandler EventHandler) {
+	if er.eventHandlers[name] != nil {
+		klog.Fatalf("Multiple event handlers with the same name %q registered on the %q registry ", name,
+			er.name)
+	}
```

Comment:
We also need to add `eventHandler` to the map.

```suggestion
        name := eventHandler.GetName()
	if er.eventHandlers[name] != nil {
		klog.Fatalf("An event handler with name %q already exists for registry %q", name, er.name)
	}
	
	er.eventHandlers[name] = eventHandler
```